airflow/contrib/hooks/gcp_dataflow_hook.py

```diff
@@ -381,17 +392,24 @@ def start_python_dataflow(self, job_name, variables, dataflow, py_options,
         :type dataflow: str
         :param py_options: Additional options.
         :type py_options: list
+        :param py_interpreter: Python version of the beam pipeline.
+            This defaults to the python version that airflow is running.
+        :type py_interpreter: str
         :param append_job_name: True if unique suffix has to be appended to job name.
         :type append_job_name: bool
         """
+        if not py_interpreter:
+            py_interpreter = 'python' + '.'.join(map(str, sys.version_info[0:2]))
+        if py_interpreter not in DATAFLOW_SUPPORTED_PY_INTERPRETERS:
```

Comment:
This check seems a bit restrictive to me and introduces the burden of backfilling latest supported python versions (e.g., 3.8). How about we go with py2/3 and let self._start_dataflow() fail by itself?

Reply:
Ok. I want to allow user to specify a minor version of interpreter if they want. But we can let `self._start_dataflow()` fail if it's unsupported version or if user specifies minor version that is not available on their system. 
1. Removed this check
1. I default to `python2` or `python3` based on major version of the `sys.version_info` of this airflow process.